Csv tools

CHARLONCyril opened the pull request:
Update the CSVUtils class so that it can write an object to a cvs file

Commit:
creation of SearchPreferences class to store user's preferences

Commit:
Adaptation of CSVUtils class for writing the content of SearchPreferences class in csv file and read csv file for implementation of SearchPreference object

Commit:
added missing dependency for univocity

Commit:
cherry-picking from CSVTools branch

oliviercailloux commented on src/main/java/io/github/oliviercailloux/twod_library/controller/CSVUtils.java:
```diff
@@ -0,0 +1,87 @@
+	 * @see <a href=
+	 *      "https://www.univocity.com/pages/univocity_parsers_writing.html#writing">univocity_parsers_writing.html#writing</a>
+	 */
+	public static void writeIntoCSVFile(String fileName, SearchPreferences... list) throws URISyntaxException {
```
Une `List` est préférée à un varargs (qui cache un bête « array »).

oliviercailloux commented on src/main/java/io/github/oliviercailloux/twod_library/controller/CSVUtils.java:
```diff
@@ -0,0 +1,87 @@
+	}
+
+	public static String getPathLocation() {
+		return "C:/twod_library/controller/";
```
Un chemin relatif est largement préférable. Et renvoyer un `Path` pour mieux décrire l’objet renvoyé.

oliviercailloux commented on src/main/resources/io/github/oliviercailloux/twod_library/controller/UserPreference.csv:
```diff
@@ -0,0 +1,2 @@
+Background_color,Shelves_Color,Books_Color,Position_of_books,Sort_books_by,Books_per_shelf
```
Nommer ce fichier `DefaultUserPreference`, si c’est bien à ça qu’il est censé servir. (Ou m’expliquer à quoi il sert.)

oliviercailloux reviewed (changes requested):
Fournir un test unitaire pour vérifier le parsing d’un CSV en SearchPreference.

Commit:
transformation of SearchPreferences class into immutable class

girardet committed:
add test file for csv parsing, renamed user's preferences file and  replace varargs by list

girardet commented:
Add test for parsing, rename csv file, change varargs to list, use path for describe object.

oliviercailloux commented on src/main/java/io/github/oliviercailloux/twod_library/controller/CSVUtils.java:
```diff
@@ -0,0 +1,87 @@
+			Path pathDestFile = Paths.get(getPathLocation() + fileName);
+			Files.copy(pathSourceFile, pathDestFile);
+		} else {
+			LOGGER.info("The file " + fileName + " already exists.");
```
Échec rapide ? D’après son nom, cette méthode est supposée fournir à son appelant une garantie que le fichier contient les bons paramètres, pas ne rien faire silencieusement quand le fichier existe déjà. Si vous voulez ne rien faire quand le fichier existe, il faut par exemple l’appeler `initializeSettingsIfNotExists`.

oliviercailloux commented on src/test/java/io/github/oliviercailloux/twod_library/controller/CSVUtilsTest.java:
```diff
@@ -0,0 +1,25 @@
+public class CSVUtilsTest {
+
+	@Test
+	public void test_parseCSVFile() throws URISyntaxException {
```
CASING

oliviercailloux commented on src/test/java/io/github/oliviercailloux/twod_library/controller/CSVUtilsTest.java:
```diff
@@ -0,0 +1,25 @@
+	@Test
+	public void test_parseCSVFile() throws URISyntaxException {
+		java.util.List<SearchPreferences> actual = CSVUtils.parseCSVFile(new File(CSVUtils.getURLSettingFile()));
+		assertEquals("AUTO", actual.get(0).getBackgroundColor().toString());
```
Il serait plus élégant de tester l’égalité des objets (couleurs, etc.) plutôt que leur représentation en `string`.

oliviercailloux commented on src/test/java/io/github/oliviercailloux/twod_library/controller/CSVUtilsTest.java:
```diff
@@ -0,0 +1,25 @@
+		assertEquals("AUTO", actual.get(0).getBookColor().toString());
+		assertEquals("LEANED", actual.get(0).getBookPosition().toString());
+		assertEquals("AUTHOR", actual.get(0).getSortType().toString());
+		assertEquals("19", String.valueOf(actual.get(0).getNbBookPerShelf()));
```
Ici il est particulièrement bizarre de convertir en String.

oliviercailloux commented on src/test/java/io/github/oliviercailloux/twod_library/controller/CSVUtilsTest.java:
```diff
@@ -0,0 +1,25 @@
+		assertEquals("LEANED", actual.get(0).getBookPosition().toString());
+		assertEquals("AUTHOR", actual.get(0).getSortType().toString());
+		assertEquals("19", String.valueOf(actual.get(0).getNbBookPerShelf()));
+	}
```
Et l’écriture ?

oliviercailloux reviewed (changes requested):
J’attends fusion de SearchPreferences, redemander review ensuite, car ici votre SearchPreferences n’est pas à jour.

CHARLONCyril committed:
merging storeUserPreferences

nces' into CSVTools

# Veuillez entrer un message de validation pour expliquer en quoi cette fusion est
# nécessaire, surtout si cela fusionne une branche amont mise à jour dans une branche de sujet.
#
# Les lignes commençant par '#' seront ignorées, et un message vide
# abandonne la validation.
